### PR TITLE
[CWS] Activity dump kernel ratelimiter improvements

### PR DESCRIPTION
**Pull request by spikat**
### What does this PR do?

Add a round robin upon four different rate limiter algorithms:
- a basic one, limiting at rate per sec (the previous one)
- an half one, limiting at rate/2 per half a sec
- one with a decreasing drop rate from 1 to 80% (after reaching rate/5 events)
- one with an increasing drop rate from 100 to 10% (after reaching rate/5 events)

### Motivation

Be able to catch (after several tries) all files that would be dropped by the simple algorithm (like with a `find /`)

### Describe how to test/QA your changes

I've modified the `TestActivityDumps/activity-dump-comm-rate-limiter`, this should be a good start.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/tests/activity_dumps_test.go [hunk 1]
```diff
@@ -229,15 +231,16 @@ func TestActivityDumps(t *testing.T) {
 
 		time.Sleep(2 * time.Second) // a quick sleep to let starts and snapshot events to be added to the dump
 
-		for i := 0; i < 20; i++ {
+		for i := 0; i < testActivityDumpRateLimiter*10; i++ {
 			if err := os.MkdirAll("/tmp/ratelimiter", os.ModePerm); err != nil {
 				t.Fatal(err)
 			}
+			defer os.Remove("/tmp/ratelimiter")
```
Comment (paulcacheux) [inline pkg/security/tests/activity_dumps_test.go:238]: The defer is executed at the end of the function, so we are pushing `testActivityDumpRateLimiter*10 ` remove of the same directory to the end of the function here. Maybe we can use `t.TempDir` https://pkg.go.dev/testing#T.TempDir ?

File: pkg/security/ebpf/c/activity_dump.h [hunk 2]
```diff
@@ -354,12 +355,95 @@ __attribute__((always_inline)) void cleanup_traced_state(u32 pid) {
...
+enum rate_limiter_algo_ids {
+    RL_ALGO_BASIC = 0,
+    RL_ALGO_BASIC_HALF,
+    RL_ALGO_DECREASING_DROPRATE,
+    RL_ALGO_INCREASING_DROPRATE,
+    RL_ALGO_TOTAL_NUMBER,
+};
+
+__attribute__((always_inline)) u8 activity_dump_rate_limiter_allow_basic(struct activity_dump_config *config, u64 now, struct activity_dump_rate_limiter_ctx* rate_ctx_p, u64 delta) {
+    if (delta > 1000000000) { // if more than 1 sec ellapsed we reset the period
```
[the remark this answers is not in the document]
Comment (spikat) [reply pkg/security/ebpf/c/activity_dump.h:374]: For sure, it's now done. Thanks!

File: pkg/security/tests/activity_dumps_test.go (hunk 1 above)
Comment (spikat) [reply pkg/security/tests/activity_dumps_test.go]: You're right .. Now it should be better, thank you!

**Review by paulcacheux — APPROVED** [review]

**Review by Gui774ume — APPROVED** [review]
Looks great ! 🎉

**Pull request merged**